### PR TITLE
ELEMENTS-523: backport upload providers to 2.2.x

### PR DESCRIPTION
**Pull request by Gabez0r**

### CODE REVIEW

File: widgets/nuxeo-uploader-behavior.html [hunk 1]
```diff
@@ -17,55 +17,244 @@
...
+      },
 
       /**
-       * List of files in the current batch.
+       * Returns all registered upload providers.
        */
-      files: {
-        type: Array,
-        value: []
+      getProviders: function() {
```
Comment (nmpcunha) [inline widgets/nuxeo-uploader-behavior.html]: Do we want to expose this?

**Comment by nmpcunha** [comment]
👍

File: widgets/nuxeo-uploader-behavior.html (hunk 1 above)
Comment (Gabez0r) [reply widgets/nuxeo-uploader-behavior.html]: For now we're only using it for debugging.

**Comment by nelsonsilva** [comment]
👍

**Pull request merged**